app/options.go

```diff
@@ -4,56 +4,69 @@
 package app
 
 import (
+	"github.com/pkg/errors"
+
 	"github.com/mattermost/mattermost-server/store"
 )
 
-type Option func(s *Server)
+type Option func(s *Server) error
 
 // By default, the app will use the store specified by the configuration. This allows you to
 // construct an app with a different store.
 //
 // The override parameter must be either a store.Store or func(App) store.Store.
 func StoreOverride(override interface{}) Option {
-	return func(s *Server) {
+	return func(s *Server) error {
 		switch o := override.(type) {
 		case store.Store:
 			s.newStore = func() store.Store {
 				return o
 			}
+			return nil
+
 		case func(*Server) store.Store:
 			s.newStore = func() store.Store {
 				return o(s)
 			}
+			return nil
+
 		default:
-			panic("invalid StoreOverride")
+			return errors.New("invalid StoreOverride")
 		}
 	}
 }
 
-func ConfigFile(file string) Option {
-	return func(s *Server) {
+func ConfigFile(file string, watch bool) Option {
+	return func(s *Server) error {
 		s.configFile = file
+		s.disableConfigWatch = !watch
```

Comment:
The var disableConfigWatch is causing a bunch of double negatives everywhere (`!disableConfigWatch`). Maybe change to enableConfigWatch? 0/5

Reply:
I'm inclined to leave this for now, given we're somewhat tied to the `--disableconfigwatch` terminology. If I modelled that as `enableConfigWatch` internally, I'd probably jump through as much hoops to ensure it starts as `true`.

That being said, I think most of these pain points go away in https://github.com/mattermost/mattermost-server/pull/10243 anyway.